### PR TITLE
Added `strictProtocolChecks` named parameter to `Peer` and `Server` constructors

### PR DESCRIPTION
**Pull request by bkonyi**

### CODE REVIEW

File: lib/src/server.dart [hunk 1]
```diff
@@ -61,17 +61,28 @@ class Server {
   /// invoked. If it is not set, the exception will be swallowed.
   final ErrorCallback onUnhandledError;
 
+  /// If true, this [Server] will accept some requests which are not conformant
+  /// with the JSON-RPC 2.0 specification. In particular, requests missing the
+  /// `jsonrpc` parameter will be accepted.
```
Comment (natebosch) [inline lib/src/server.dart:66]: This looks backwards to me. We should also format with a header.
```suggestion
  /// Whether to strictly enforce the JSON-RPC 2.0 specification for received messages.
  ///
  /// If `false`, this [Server] will accept some requests which are not conformant
  /// with the JSON-RPC 2.0 specification. In particular, requests missing the
  /// `jsonrpc` parameter will be accepted.
```

Comment (bkonyi) [reply lib/src/server.dart]: Done. Not sure what you mean about "format with a header"?

Comment (natebosch) [reply lib/src/server.dart]: I mean having a summary sentence, then a blank line.

https://dart.dev/guides/language/effective-dart/documentation#do-separate-the-first-sentence-of-a-doc-comment-into-its-own-paragraph

**Comment by natebosch** [comment]
This is now published and tagged.

**Pull request merged**